src/scheduled/weeklyStatistics.ts

```diff
@@ -1,166 +1,167 @@
-import { EmbedBuilder, ForumChannel, Guild, GuildForumTag, TextChannel } from 'discord.js';
-import { Client } from '../types';
-import { getDefaultEmbed } from '../utils/embeds.js';
+import type { EmbedBuilder, ForumChannel, Guild, GuildForumTag, TextChannel, Client } from 'discord.js'
+import { getDefaultEmbed } from '../utils/embeds.js'
 
 const getTagName = async (guild: Guild, fullTagList: GuildForumTag[], id: string) => {
-	const forumTag = fullTagList.find((forumTag) => forumTag.id == id);
+	const forumTag = fullTagList.find((tag) => tag.id === id)
 
-	let emoji = '';
+	let emoji = ''
 
 	if (forumTag) {
 		if (forumTag.emoji) {
 			if (forumTag.emoji.id) {
-				const guildEmoji = await guild.emojis.fetch(forumTag.emoji.id);
-				emoji = `<:${guildEmoji.name}:${guildEmoji.id}> `;
+				const guildEmoji = await guild.emojis.fetch(forumTag.emoji.id)
+				emoji = `<:${guildEmoji.name}:${guildEmoji.id}> `
 			} else {
-				emoji = `${forumTag.emoji.name!} `;
+				emoji = `${forumTag.emoji.name!} `
 			}
 		}
 
-		return `${emoji}${forumTag?.name}`;
+		return `${emoji}${forumTag?.name}`
 	}
-};
+}
 
 export default {
 	time: process.env.STATS_SCHEDULE,
 	async execute(client: Client) {
 		if (!process.env.GUILD_ID) {
-			console.warn('No GUILD_ID enviroment variable was set. Skipping weekly statistics');
-			return;
+			console.warn('No GUILD_ID enviroment variable was set. Skipping weekly statistics')
+			return
 		}
 
-		const guild = await client.guilds.fetch(process.env.GUILD_ID);
+		const guild = await client.guilds.fetch(process.env.GUILD_ID)
 
 		if (!process.env.SUPPORT_CHANNEL) {
-			console.warn('No SUPPORT_CHANNEL enviroment variable was set. Skipping weekly statistics');
-			return;
+			console.warn('No SUPPORT_CHANNEL enviroment variable was set. Skipping weekly statistics')
+			return
 		}
 
-		const forum: ForumChannel = (await guild.channels.fetch(process.env.SUPPORT_CHANNEL)) as ForumChannel;
+		const forum: ForumChannel = (await guild.channels.fetch(process.env.SUPPORT_CHANNEL)) as ForumChannel
 
-		const date = new Date();
+		const lastInterval = new Date()
+		lastInterval.setDate(lastInterval.getDate() - 7)
 
-		date.setMonth(date.getMonth() - 1);
+		const _threads = (await Promise.all([forum.threads.fetch(), forum.threads.fetchArchived()])).map(t=>t.threads.filter((x) => x.createdAt! > lastInterval))
+		const threads = [..._threads[0].values(), ..._threads[1].values()]
```

Comment:
I think this should be `[forum.threads.fetchActive(), forum.threads.fetchArchived()]` (only active, not all threads) because otherwise `threads` will contain duplicates, right? Since results from the first fetch and second fetch can overlap.